src/dotnet/CarbonAware.Aggregators/src/CarbonAware/CarbonAwareAggregator.cs

```diff
@@ -3,66 +3,83 @@
+    {
+        ValidateAverageProps(props);
+        var list = await GetEmissionsDataAsync(props);
+        var value = list.Any() ? list.Select(x => x.Rating).Average() : 0;
+        _logger.LogInformation($"Carbon Intensity Average: {value}");
+        return value;
+    }
 
-        public async Task<IEnumerable<EmissionsData>> GetEmissionsDataAsync(IDictionary props)
-        {
-            DateTimeOffset end = GetOffsetOrDefault(props, CarbonAwareConstants.End, DateTimeOffset.Now);
-            DateTimeOffset start = GetOffsetOrDefault(props, CarbonAwareConstants.Start, end.AddDays(-7));
-            _logger.LogInformation("Aggregator getting carbon intensity from data source");
-            return await this._dataSource.GetCarbonIntensityAsync(GetOrDefaultLocation(props), start, end);
+    /// <summary>
+    /// Extracts the given offset prop and converts to DateTimeOffset. If prop is not defined, defaults to provided default
+    /// </summary>
+    /// <param name="props"></param>
+    /// <returns>DateTimeOffset representing end period of carbon aware data search. </returns>
+    /// <exception cref="ArgumentException">Throws exception if prop isn't a valid DateTimeOffset. </exception>
+    private DateTimeOffset GetOffsetOrDefault(IDictionary props, string field, DateTimeOffset defaultDto) 
+    {
+        var dto = props[field];
 
+        // If null, default
+        if (dto == null)
+        {
+            return defaultDto;
         }
-
-        /// <summary>
-        /// Extracts the given offset prop and converts to DateTimeOffset. If prop is not defined, defaults to provided default
-        /// </summary>
-        /// <param name="props"></param>
-        /// <returns>DateTimeOffset representing end period of carbon aware data search. </returns>
-        /// <exception cref="ArgumentException">Throws exception if prop isn't a valid DateTimeOffset. </exception>
-        private DateTimeOffset GetOffsetOrDefault(IDictionary props, string field, DateTimeOffset defaultDto) 
+        // If fail to parse property, throw error
+        if (!DateTimeOffset.TryParse(dto.ToString(), out defaultDto))
         {
-            var dto = props[field];
+            Exception ex = new ArgumentException("Failed to parse" + field + "field. Must be a valid DateTimeOffset");
+            _logger.LogError("argument exception", ex);
+            throw ex;
+        }
 
-            // If null, default
-            if (dto == null)
-            {
-                return defaultDto;
-            }
-            // If fail to parse property, throw error
-            if (!DateTimeOffset.TryParse(dto.ToString(), out defaultDto))
-            {
-                Exception ex = new ArgumentException("Failed to parse" + field + "field. Must be a valid DateTimeOffset");
-                _logger.LogError("argument exception", ex);
-                throw ex;
-            }
+        return defaultDto;
+    }
 
-            return defaultDto;
```

Comment:
The method name - GetOrDefautlLocation is misleading. We are not setting a default Location if not present but throwing an Exception. Perhaps change it to GetLocationOrThrow?

Reply:
Whoops, that's coming from my PR that he's pulling in, I'll fix upstream